addons/mail/static/src/composer/suggestion_hook.js

```diff
@@ -114,7 +122,7 @@ export function useSuggestion() {
             );
             const { type, mainSuggestions, extraSuggestions = [] } = suggestions;
             if (!mainSuggestions.length && !extraSuggestions.length) {
-                self.state.items = undefined;
+                self.clearSearch();
```

Comment:
Did you need to change this one? I specifically didn't do it in my PR because it was breaking something else at some point (possibly when using backspace to erase chars when there is no result to get back previous result, possibly when there is a space in the content) and ended up clearing inside `onFetched` instead.

Reply:
Yes, this is the fix for what was requested. I also added the `onFetched` part since it was in your PR and leads to similar errors. There is a test that covers the "erase char" part, it still passes and I also tried manually, it looks OK. Same for the "space in content" scenario.

Reply:
But if `update` is already doing full clear inside it now, I believe the `onFetched` clear just after `update` is not necessary.

Reply:
When we are good to go, just to make sure I'll have to run this fix on my PR in master so see why I thought this was not the correct way.